.gitignore

```diff
@@ -0,0 +1,8 @@
+/scratchpad.ipynb
+/data/*
+__pycache__
+.pytest_cache
+.vscode
+notebook.ipynb
+notes.txt
+softmax_derivative.ipynb
```

Comment:
Eine kleine Marotte von mir: Leg in der *.gitignore* nur Dateipfade ab, die einer normaler Nutzer deines Codes oder einer deiner Kollegen auch ignorieren sollte. Wenn du eigene "Schmierzettel" ignorieren möchtest, nutze dafür die *.git/info/exclude*, eine lokale ignore-Datei die den gleichen Effekt besitzt, aber nicht synchronisiert wird.

Du kannst dir vielleicht ausmalen, wie die *.gitignore* aussieht, wenn ein Team aus fünf Leuten anfängt ihre Notizdateien, Prototypnotebooks und Testcode hier aufzulisten:

```
foo.py
asdf.py
fragen.md
test.ipynb
Test_2.ipynb
I'm a barbie girl.mp3
```

Als grunsätzlich: Typischerweise auftretende Dateien (z.B. .vscode/, pycache, ...) in *.gitignore*, deine eigenen Dateinen in die *.git/info/exclude*.